.github/workflows/reusable.yml

```diff
@@ -0,0 +1,104 @@
+name: Deploy AKS-Construction
+
+on:
+  workflow_call:
+    inputs:
+      templateVersion:
+        description: 'Template Version'
+        required: false
+        default: '0.8.5-preview'
+        type: string
+      rg:
+        description: 'ResourceGroup name'
+        required: true
+        type: string
+      resourceName:
+        description: 'Unique name all your resources are based on'
+        required: true
+        type: string
+      templateParamFile:
+        description: 'Template Parameters file'
+        required: false
+        type: string
+      templateParams:
+        description: 'Template Paramterss, space seperated key=value'
+        required: true
+        type: string
+      postScriptParams:
+        description: 'Template Paramterss, comma seperated key=value'
+        required: false
+        type: string
+      postScriptInvokeCommand:
+        description: 'Private Cluster, so use invoke commant for postscript'
+        required: false
+        default: false
+        type: boolean
+    secrets:
+      AZURE_CLIENT_ID:
+        required: true
+      AZURE_TENANT_ID:
+        required: true
+      AZURE_SUBSCRIPTION_ID:
+        required: true
+      USER_OBJECT_ID:
+        required: false
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  aksc_reusable_workflow:
```

Comment:
you're not setting environment on the job. i think this is needed in order to support GitHub environments

Reply:
#316  fixed